pkg/status/combinedstatus-resolver.go

```diff
@@ -242,9 +242,6 @@ func (c *combinedStatusResolver) NoteBindingResolution(bindingName string, bindi
 			c.resolutionNameToKey[csResolution.getName()] = resolutionKey{bindingName, objectIdentifier}
 		}
 
-		// fetch missing statuscollector specs
-		c.fetchMissingStatusCollectorSpecsLocked(statusCollectorLister, objectData.StatusCollectors)
```

Comment:
Missing collectors should be pending/about to be pending in the workqueue. Removal is simpler than validating them here as well.